File: app/src/main/java/com/blues/model_wanandroid/netease/CourseEntryHolder.java
package com.blues.model_wanandroid.netease;

import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.blues.R;
import com.blues.database.course.CourseEntity;
import com.blues.framework.utils.ScreenUtil;
import com.blues.framework.utils.SizeUtil;
import com.bumptech.glide.Glide;
import com.jude.easyrecyclerview.adapter.BaseViewHolder;

/**
 * User : Blues
 * Date : 2019/8/8
 * Time : 15:49
 */

public class CourseEntryHolder extends BaseViewHolder<CourseEntity> {

    private ImageView ivIcon;
    private TextView tvTitle;

    public CourseEntryHolder(ViewGroup parent) {
        super(parent, R.layout.item_course_entry);

        ivIcon = $(R.id.iv_entry_icon);
        tvTitle = $(R.id.tv_entry_title);
        LinearLayout llContainer = $(R.id.ll_course_entry);

        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT);
        params.width = ScreenUtil.getScreenWidth() / 4;
        params.topMargin = SizeUtil.dp2px(5);
        params.bottomMargin = SizeUtil.dp2px(5);

        llContainer.setLayoutParams(params);
    }

    @Override
    public void setData(CourseEntity data) {
        super.setData(data);

        Glide.with(getContext()).load(data.getIcon()).into(ivIcon);
        tvTitle.setText(data.getName());
    }
}

File: app/src/main/java/com/blues/home/view/CourseEntryHolder.kt
package com.blues.home.view

import com.blues.framework.utils.ScreenUtil.screenWidth
import android.view.ViewGroup
import android.widget.ImageView
import com.blues.bean.CourseEntity
import com.blues.R
import android.widget.TextView
import android.widget.LinearLayout
import coil.load
import com.blues.framework.utils.dp
import com.jude.easyrecyclerview.adapter.BaseViewHolder

/**
 * File: com.blues.module_main.CourseEntryHolder
 * Description: xxx
 *
 * @author: lanhuajian
 * @time: 2021-07-16
 */
class CourseEntryHolder(parent: ViewGroup?) :
    BaseViewHolder<CourseEntity>(parent, R.layout.item_course_entry) {

    private val tvTitle: TextView = itemView.findViewById(R.id.tv_entry_title)

    private val ivIcon: ImageView = itemView.findViewById(R.id.iv_entry_icon)

    override fun setData(data: CourseEntity) {
        super.setData(data)

        ivIcon.load(data.icon)

        tvTitle.text = data.name
    }

    init {
        val llContainer = itemView.findViewById<LinearLayout>(R.id.ll_course_entry)
        LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT)
                .apply {
                    width = screenWidth / 4
                    topMargin = 5f.dp.toInt()
                    bottomMargin = 5f.dp.toInt()
                }
                .also {
                    llContainer.layoutParams = it
                }
    }
}